src/base/Roles/CrossChain/MultiChainHyperlaneTellerWithMultiAssetSupport.sol

```diff
@@ -87,11 +87,14 @@ contract MultiChainHyperlaneTellerWithMultiAssetSupport is MultiChainTellerBase
      * @param data Bridge data
      */
     function _quote(uint256 shareAmount, BridgeData calldata data) internal view override returns (uint256) {
-        bytes memory _payload = abi.encode(shareAmount, data.destinationChainReceiver);
+        bytes32 messageId = keccak256(abi.encodePacked(++nonce, address(this), block.chainid));
```

Comment:
May be worth wrapping this in an unchecked block to save some gas on nonce increment